subsys/bluetooth/audio/ascs.c

```diff
@@ -1733,28 +1733,59 @@ static void ase_qos(struct bt_ascs_ase *ase, const struct bt_ascs_qos *qos)
 	ascs_cp_rsp_success(ASE_ID(ase), BT_ASCS_QOS_OP);
 }
 
+static bool ascs_qos_check_len(struct net_buf_simple *buf)
+{
+	const struct bt_ascs_qos_op *op;
+	struct net_buf_simple_state state;
+
+	net_buf_simple_save(buf, &state);
+
+	if (buf->len < sizeof(*op)) {
+		LOG_WRN("Malformed PDU");
+		return false;
+	}
+
+	op = net_buf_simple_pull_mem(buf, sizeof(*op));
+	if (op->num_ases < 1) {
+		LOG_WRN("Number_of_ASEs parameter value is less than 1");
+		return false;
+	}
+
+	for (uint8_t i = 0U; i < op->num_ases; i++) {
```

Comment:
that loop could be avoided

if (buf->len < sizeof(*qos) * op->num_ases) { return false;};

net_buf_simple_pull_mem(buf, sizeof(*qos) * op->num_ases);